### PR TITLE
added balance listener and subscribe from dashboard to send the balan…

### PR DESCRIPTION
**Pull request by bradleySuira**
Added a helper and make polling every n seconds to get both balances (vtho and eth) updated and send through the child components

### CODE REVIEW

File: src/Components/Base/BalanceListener.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        this.createBalanceSubscription((window as any).web3Object.eth.getBalance(this._web3DefaultAccount),
+        listener)
+        return this
+    }
+
+    public stop(): void {
+        this._subscriptions$.forEach(sub$ => sub$.unsubscribe())
+    }
+
+    private createBalanceSubscription(
```
Comment (molekilla) [inline src/Components/Base/BalanceListener.ts]: getWalletGasBalances

File: src/Components/Base/BalanceListener.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+
+
+import { Subscription, interval, from, of } from 'rxjs'
+import { flatMap, switchMap } from 'rxjs/operators'
+const LISTENER_INTERVAL = 3000
+
+export default class BalanceListener {
+    private _subscriptions$: Subscription[] = []
```
Comment (molekilla) [inline src/Components/Base/BalanceListener.ts]: Array no longer needed, will be  a a `subscription: Subscription`

File: src/Components/Dashboard/Dashboard.jsx [hunk 3]
```diff
@@ -92,55 +101,21 @@ class Dashboard extends Component {
...
-        // })
-    }
 
-    initVTHOBalance = async () => {
-        try {
-            // VET balance
-            const balance = await window.thor.eth.getEnergy(
-                window.thor.eth.defaultAccount
-            )
+    loadBalances = ()=> {
```
Comment (molekilla) [inline src/Components/Dashboard/Dashboard.jsx]: Will call getWalletGasBalances

File: src/Components/Wallet/Wallet.jsx [hunk 4]
```diff
@@ -194,6 +194,19 @@ class Wallet extends Component {
     }
 
     initWeb3Data = () => {
+        let balances = this.state.balances
```
Comment (molekilla) [inline src/Components/Wallet/Wallet.jsx]: Verify it matches the state model located in WalletState

File: src/Components/Base/BalanceListener.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
+
+
+import { Subscription, interval, from, of } from 'rxjs'
+import { flatMap, switchMap } from 'rxjs/operators'
+const LISTENER_INTERVAL = 3000
```
[the remark this answers is not in the document]
Comment (bradleySuira) [reply src/Components/Base/BalanceListener.ts]: done, set in 8 as we talk

[the remark this answers is not in the document]
Comment (bradleySuira) [reply src/Components/Base/BalanceListener.ts]: done

File: src/Components/Base/BalanceListener.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+export default class BalanceListener {
+    private _subscriptions$: Subscription[] = []
+    private _web3DefaultAccount: string
+    private _thorDefaultAccount: string
+
+     constructor() {
```
[the remark this answers is not in the document]
Comment (bradleySuira) [reply src/Components/Base/BalanceListener.ts]: done

File: src/Components/Base/BalanceListener.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    ) {
+        const subscription$ = interval(LISTENER_INTERVAL).pipe(
+            flatMap(() => from(promise)),
+            switchMap(i => of(i))
+        ).subscribe(balance => {
+            console.log(`----------listenForBalance:`, balance)
+            listener(balance)
+        })
+
+        this._subscriptions$.push(subscription$)
```
[the remark this answers is not in the document]
Comment (bradleySuira) [reply src/Components/Base/BalanceListener.ts]: done

File: src/Components/Base/BalanceListener.ts (hunk 2 above)
Comment (bradleySuira) [reply src/Components/Base/BalanceListener.ts]: done

**Pull request merged**